quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -230,6 +234,33 @@ void configureProfile(KeycloakRecorder recorder) {
         recorder.configureProfile(profile.getName(), profile.getFeatures());
     }
 
+    /**
+     * Check whether JDBC driver is present for the specified DB
+     *
+     * @param ignore used for changing build items execution order with regards to AgroalProcessor
+     */
+    @BuildStep(onlyIf = IsJpaStoreEnabled.class)
+    @Produce(CheckJdbcBuildStep.class)
+    void checkJdbcDriver(BuildProducer<JdbcDriverBuildItem> ignore) {
+        final Optional<String> dbVendor = Configuration.getOptionalValue(NS_KEYCLOAK_PREFIX.concat(DatabaseOptions.DB.getKey()));
+
+        if (dbVendor.isPresent()) {
+            boolean xaEnabled = Configuration.getOptionalValue(NS_KEYCLOAK_PREFIX.concat(TRANSACTION_XA_ENABLED.getKey()))
+                    .map("xa"::equals)
+                    .orElseGet(() -> TRANSACTION_XA_ENABLED.getDefaultValue().orElse(true));
+
+            final String dbDriver = Database.getDriver(dbVendor.get(), xaEnabled)
+                    .orElseThrow(() -> new ConfigurationException(String.format("Unknown DB vendor '%s'", dbVendor.get())));
```

Comment:
Maybe we could simply fetch `quarkus.datasource.jdbc.driver` option here instead of calculating it again?

Reply:
@vmuzikar Good point, thanks.